lib/progress.ts

```diff
@@ -129,16 +129,46 @@ export const getLastUnlockedLessonPath = (userProgressKey: string): string => {
   return path
 }
 
-export const getNextLessonKey = (userProgressKey: string): string => {
+export const getCurrentLessonKey = (
+  userProgressKey: string,
+  account?: any
+): string => {
+  if (!account) {
+    return userProgressKey
+  }
+
   const id = keys.indexOf(userProgressKey)
-  const result = keys[id + 1]
+  const nextChapterIntro = 'CH' + getNextChapterNumber(userProgressKey) + 'INT1'
 
-  if (!result) {
+  if (keys[id + 1] && keys[id + 1] === nextChapterIntro) {
+    return nextChapterIntro
+  }
+
+  return userProgressKey
+}
+
+export const getNextLessonKey = (
+  userProgressKey: string,
+  account?: any
+): string => {
+  const id = keys.indexOf(userProgressKey)
+  const nextChapterIntro = 'CH' + getNextChapterNumber(userProgressKey) + 'INT1'
+
+  if (account && keys[id + 2] && keys[id + 2] === nextChapterIntro) {
+    return nextChapterIntro
+  }
+
+  if (keys[id + 1]) {
+    return keys[id + 1]
+  } else {
     console.error('There is no next lesson')
     return userProgressKey
   }
+}
 
-  return result
+export const getNextChapterNumber = (userProgressKey: string): number => {
+  const curLessonNumber = parseInt(userProgressKey.slice(2, 3))
```

Comment:
lets just rename this `currentLessonNumber` for clarity

Reply:
Addressed in ad7d8e0775e0c4948e19f3e9f25d8b3844b82cf2

Thanks for the suggestion!